third_party/2and3/chardet/universaldetector.pyi

```diff
@@ -1,19 +1,39 @@
-from typing import Dict, Union, AnyStr, Pattern
+from typing import Dict, Union, AnyStr, Pattern, Optional
 from logging import Logger
 
-class UniversalDetector:
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
```

Comment:
While `TypedDict` is only available in `typing` starting in Python 3.8, you can just import it from `typing_extensions` for earlier Python versions:

```python
if sys.version_info >= (3, 8):
    from typing import TypedDict
else:
    from typing_extensions import TypedDict
```

Type checkers support this.

Reply:
Even easier, you can also just do `from typing_extensions import TypedDict` so you don't need the version check.

Reply:
@srittau @JelleZijlstra  Thanks for that. So`typing_extensions` has the required info anyway?

Reply:
@JelleZijlstra Stubs of some other modules seems to be doing the version check as @srittau mentioned. As in 

    if sys.version_info >= (3, 8):
        from typing import Final
    else:
        from typing_extensions import Final

Does this method have any advantage?

Reply:
No advantage, except that whenever 3.7 support ends, a simple grep for "version_info" will flag this. But we should just grep for typing_extensions when that happens.

Reply:
You mean when `typing_extensions` stop supporting 3.7?

Reply:
No, when typeshed stops supporting 3.7 in a few years, we can just do `from typing import Final` in typeshed.